src/rpc/quorums.cpp

```diff
@@ -1036,6 +1036,60 @@ static UniValue submitchainlock(const JSONRPCRequest& request)
     return true;
 }
 
+static void dkginfo_help(const JSONRPCRequest& request)
+{
+    RPCHelpMan{"dkginfo",
+               "Return information regarding DKGs.\n"
+               "Enabled only for Masternode and works only when SPORK_17_QUORUM_DKG_ENABLED spork is ON.\n",
+               {
+                       {},
+               },
+               RPCResults{},
+               RPCExamples{""},
+    }.Check(request);
+}
+
+static UniValue dkginfo(const JSONRPCRequest& request)
+{
+    dkginfo_help(request);
+
+    if (!fMasternodeMode) {
+        throw JSONRPCError(RPC_INVALID_REQUEST, "RPC allowed only for Masternodes");
+    }
+
+    const NodeContext& node = EnsureAnyNodeContext(request.context);
+    const ChainstateManager& chainman = EnsureChainman(node);
+    const LLMQContext& llmq_ctx = EnsureLLMQContext(node);
+
+    llmq::CDKGDebugStatus status;
+    llmq_ctx.dkg_debugman->GetLocalDebugStatus(status);
+    UniValue ret(UniValue::VOBJ);
+    ret.pushKV("nActiveDKGs", (int)status.sessions.size());
+
+    if (status.sessions.empty()) {
+        int nTipHeight = WITH_LOCK(cs_main, return chainman.ActiveChain().Height());
+        const Consensus::Params &consensus_params = Params().GetConsensus();
+        int minDkgWindow = std::numeric_limits<int>::max();
+        for (const auto &params: consensus_params.llmqs) {
+            if (!params.useRotation)
+                minDkgWindow = std::min(minDkgWindow, params.dkgInterval - (nTipHeight % params.dkgInterval));
+            else {
+                if (nTipHeight % params.dkgInterval > params.signingActiveQuorumCount) {
+                    // Next potential DKG is the DKG for quorumIndex 0
+                    minDkgWindow = std::min(minDkgWindow, params.dkgInterval - (nTipHeight % params.dkgInterval));
+                }
+                // We are currently during a rotation cycle. Since sessions.empty(), next we return next potential DKG is 1 (next block for next quorumIndex)
+                else {
+                    minDkgWindow = std::min(minDkgWindow, 1);
+                }
+            }
```

Comment:
I believe it'd be better to refactor this into a lambda that returns minDkgWindow

Reply:
this was not done